packages/extensions/src/select-block/select-block-keymap.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  defineKeymap,
+  Priority,
+  withPriority,
+  type PlainExtension,
+} from '@prosekit/core'
+import {
+  AllSelection,
+  type Command,
+} from '@prosekit/pm/state'
+
+import {
+  isBlockSelected,
+  selectCurrentBlock,
+} from './select-block-utils'
+
+/**
+ * Ctrl-A command: first press selects current block, if block is already selected then select all
+ */
+const selectAllOrBlockCommand: Command = (state, dispatch) => {
+  const { selection } = state
+
+  // If already selected all, do nothing
+  if (selection instanceof AllSelection) {
+    return true
+  }
+
+  // If current block is already selected (but not all), then select all
+  if (isBlockSelected(selection)) {
+    if (dispatch) {
+      dispatch(state.tr.setSelection(new AllSelection(state.doc)))
+    }
+    return true
```

Comment:
We can simply return false here and allow the [original `Mod-a` keybinding](https://github.com/ProseMirror/prosemirror-commands/blob/6d6e95b9896c6f04abc5b990ccb356920344243c/src/commands.ts#L756) to manage the behavior. This approach also better accommodates the situation where some users have a different keybinding for `Mod-a`.